File: apps/api/internal/domain/cliente.go
package domain

import "time"

// Cliente es la entidad de dominio del cliente de la cafetería, identificado por DNI.
// El contador de infusiones se mantiene denormalizado para lectura rápida y habilita
// la evaluación de beneficios.
type Cliente struct {
	ID                 string
	DNI                string
	Nombre             string
	Email              *string
	InstitucionID      *string
	ContadorInfusiones int
	CreatedAt          time.Time
}

File: packages/types/src/cliente.ts
import type { ISODateString, Rol, UUID } from './common';

export interface Institucion {
  id: UUID;
  nombre: string;
  activa: boolean;
  createdAt: ISODateString;
}

export interface Cliente {
  id: UUID;
  dni: string;
  nombre: string;
  email: string | null;
  institucionId: UUID | null;
  contadorInfusiones: number;
  createdAt: ISODateString;
}

export interface Usuario {
  id: UUID;
  email: string;
  rol: Rol;
  activo: boolean;
}
